### ISSUE TITLE
Can't measure AxisPanel with Infinity.

### ISSUE THREAD

**Issue by HakamFostok** [xceedsoftware/mauitoolkit#3]
Reproducing Repo https://github.com/HakamFostok/XceedExceptionReproduce

I am using .NET 8
Xceed.Product.Maui.Toolkit.Full Version=1.5.23517.727

The exception is thrown [from this line of code](https://github.com/xceedsoftware/mauitoolkit/blob/0c542b1f8027ea6c857a4b7b348d330e35826c74/Toolkit/Src/Xceed.Maui.Toolkit/Chart/Implementation/Panels/AxisPanel.cs#L407)

Error Message
> Can't measure AxisPanel with Infinity.

StackTrace

```
   at Xceed.Maui.Toolkit.AxisPanelLayoutManager.Measure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.Layout.CrossPlatformMeasure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Platform.MauiPanel.MeasureOverride(Size availableSize)
   at ABI.Microsoft.UI.Xaml.IFrameworkElementOverrides.Do_Abi_MeasureOverride_0(IntPtr thisPtr, Size availableSize, Size* result)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler(IViewHandler viewHandler, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.LayoutExtensions.ComputeDesiredSize(IView view, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.VisualElement.MeasureOverride(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCell(Cell cell, Double width, Double height)
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.FirstMeasurePass()
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCells()
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure..ctor(IGridLayout grid, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.GridLayoutManager.Measure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.Layout.CrossPlatformMeasure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Platform.MauiPanel.MeasureOverride(Size availableSize)
   at ABI.Microsoft.UI.Xaml.IFrameworkElementOverrides.Do_Abi_MeasureOverride_0(IntPtr thisPtr, Size availableSize, Size* result)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler(IViewHandler viewHandler, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.LayoutExtensions.ComputeDesiredSize(IView view, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.VisualElement.MeasureOverride(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCell(Cell cell, Double width, Double height)
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.FirstMeasurePass()
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCells()
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure..ctor(IGridLayout grid, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.GridLayoutManager.Measure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.Layout.CrossPlatformMeasure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Platform.MauiPanel.MeasureOverride(Size availableSize)
   at ABI.Microsoft.UI.Xaml.IFrameworkElementOverrides.Do_Abi_MeasureOverride_0(IntPtr thisPtr, Size availableSize, Size* result)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler(IViewHandler viewHandler, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.LayoutExtensions.ComputeDesiredSize(IView view, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.VisualElement.MeasureOverride(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCell(Cell cell, Double width, Double height)
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.SecondMeasurePass()
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure.MeasureCells()
   at Microsoft.Maui.Layouts.GridLayoutManager.GridStructure..ctor(IGridLayout grid, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.GridLayoutManager.Measure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.Layout.CrossPlatformMeasure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Platform.MauiPanel.MeasureOverride(Size availableSize)
   at ABI.Microsoft.UI.Xaml.IFrameworkElementOverrides.Do_Abi_MeasureOverride_0(IntPtr thisPtr, Size availableSize, Size* result)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at Microsoft.Maui.ViewHandlerExtensions.GetDesiredSizeFromHandler(IViewHandler viewHandler, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Layouts.LayoutExtensions.ComputeDesiredSize(IView view, Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.ContentPresenter.MeasureOverride(Double widthConstraint, Double heightConstraint)
   at Xceed.Maui.Toolkit.BorderShapeGridLayoutManager.Measure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Controls.Layout.CrossPlatformMeasure(Double widthConstraint, Double heightConstraint)
   at Microsoft.Maui.Platform.MauiPanel.MeasureOverride(Size availableSize)
   at ABI.Microsoft.UI.Xaml.IFrameworkElementOverrides.Do_Abi_MeasureOverride_0(IntPtr thisPtr, Size availableSize, Size* result)
--- End of stack trace from previous location ---
   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
   at ABI.Microsoft.UI.Xaml.IUIElementMethods.UpdateLayout(IObjectReference _obj)
   at Microsoft.UI.Xaml.UIElement.UpdateLayout()
   at Microsoft.Maui.Controls.Handlers.Compatibility.ListViewRenderer.<OnCollectionChanged>b__23_0()
   at Microsoft.Maui.Controls.DispatcherExtensions.DispatchIfRequired(IDispatcher dispatcher, Action action)
   at Microsoft.Maui.Controls.Handlers.Compatibility.ListViewRenderer.OnCollectionChanged(Object sender, NotifyCollectionChangedEventArgs e)
   at Microsoft.Maui.Controls.Internals.TemplatedItemsList`2.OnCollectionChanged(NotifyCollectionChangedEventArgs e)
   at Microsoft.Maui.Controls.Internals.TemplatedItemsList`2.OnProxyCollectionChanged(Object sender, NotifyCollectionChangedEventArgs e)
   at Microsoft.Maui.Controls.Internals.TemplatedItemsList`2.OnItemsSourceChanged(Boolean fromGrouping)
   at Microsoft.Maui.Controls.Internals.TemplatedItemsList`2.BindableOnPropertyChanged(Object sender, PropertyChangedEventArgs e)
   at Microsoft.Maui.Controls.Element.OnPropertyChanged(String propertyName)
   at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, SetterSpecificity specificity, Boolean silent)
   at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes, SetterSpecificity specificity)
   at Microsoft.Maui.Controls.BindingExpression.ApplyCore(Object sourceObject, BindableObject target, BindableProperty property, Boolean fromTarget, SetterSpecificity specificity)
   at Microsoft.Maui.Controls.BindingExpression.Apply(Boolean fromTarget)
   at Microsoft.Maui.Controls.BindingExpression.BindingExpressionPart.<PropertyChanged>b__50_0()
   at Microsoft.Maui.Controls.DispatcherExtensions.DispatchIfRequired(IDispatcher dispatcher, Action action)
   at Microsoft.Maui.Controls.BindingExpression.WeakPropertyChangedProxy.OnPropertyChanged(Object sender, PropertyChangedEventArgs e)
   at Microsoft.Maui.Controls.Element.OnPropertyChanged(String propertyName)
   at Xceed.Maui.Toolkit.Control.OnPropertyChanged(String propertyName)
   at Microsoft.Maui.Controls.BindableObject.SetValueActual(BindableProperty property, BindablePropertyContext context, Object value, Boolean currentlyApplying, SetValueFlags attributes, SetterSpecificity specificity, Boolean silent)
   at Microsoft.Maui.Controls.BindableObject.SetValueCore(BindableProperty property, Object value, SetValueFlags attributes, SetValuePrivateFlags privateAttributes, SetterSpecificity specificity)
   at Microsoft.Maui.Controls.BindableObject.SetValue(BindableProperty property, Object value)
   at Xceed.Maui.Toolkit.Legend.set_ItemsSource(IEnumerable value)
   at Xceed.Maui.Toolkit.Chart.UpdateLegend()
   at Xceed.Maui.Toolkit.Chart.Refresh(Boolean refreshAll)
   at Xceed.Maui.Toolkit.Chart.Chart_Loaded(Object sender, EventArgs e)
   at Microsoft.Maui.Controls.VisualElement.SendLoaded(Boolean updateWiring)
   at Microsoft.Maui.Controls.VisualElement.SendLoaded()
   at Microsoft.Maui.Platform.FrameworkElementExtensions.<>c__DisplayClass18_0.<OnLoaded>b__2(Object _, RoutedEventArgs __)
   at ABI.Microsoft.UI.Xaml.RoutedEventHandler.Do_Abi_Invoke(IntPtr thisPtr, IntPtr sender, IntPtr e)
```

Steps to reproduce

I copied the code from the read me file (without any changes) I copied it from [here](https://github.com/xceedsoftware/mauitoolkit/tree/0c542b1f8027ea6c857a4b7b348d330e35826c74#chart)

**Comment by XceedBoucherS** [comment]
Hi @HakamFostok 
I copied the code you pointed out in a new NET8 MAUI project.
I downloaded the Xceed.Product.Maui.Toolkit.Full v1.5 from NuGet(the latest).
I didn't have any exception in Windows nor Android. 
Which platform are you using ?

It looks like a size not set for the Chart.

Here's my MainPage.xaml:
`<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:xctk="http://schemas.xceed.com/maui/xaml/toolkit"
             x:Class="MauiApp11.MainPage">

  <xctk:Chart x:Name="MyChart">
    <xctk:Chart.HorizontalAxis>
      <xctk:Axis TickLabelType="Text" />
    </xctk:Chart.HorizontalAxis>
    <xctk:Chart.Series>
      <xctk:Series x:Name="FirstSeries">
        <xctk:Series.DataPoints>
          <xctk:DataPoint Text="0"
                          Y="44" />
          <xctk:DataPoint Text="1"
                          Y="64" />
          <xctk:DataPoint Text="2"
                          Y="60" />
          <xctk:DataPoint Text="3"
                          Y="32" />
          <xctk:DataPoint Text="4"
                          Y="52" />
          <xctk:DataPoint Text="5"
                          Y="72" />
          <xctk:DataPoint Text="6"
                          Y="32" />
        </xctk:Series.DataPoints>
      </xctk:Series>
    </xctk:Chart.Series>
  </xctk:Chart>

</ContentPage>`

And here my MauiProgram.cs:
`using Microsoft.Extensions.Logging;
using Xceed.Maui.Toolkit;

namespace MauiApp11
{
  public static class MauiProgram
  {
    public static MauiApp CreateMauiApp()
    {
      var builder = MauiApp.CreateBuilder();
      builder
        .UseMauiApp<App>()
        .UseXceedMauiToolkit()
        .ConfigureFonts( fonts =>
        {
          fonts.AddFont( "OpenSans-Regular.ttf", "OpenSansRegular" );
          fonts.AddFont( "OpenSans-Semibold.ttf", "OpenSansSemibold" );
        } );

#if DEBUG
		builder.Logging.AddDebug();
#endif

      return builder.Build();
    }
  }
}`

**Comment by HakamFostok** [comment]
@XceedBoucherS 
I created a repository to reproduce it, here is it https://github.com/HakamFostok/XceedExceptionReproduce
please clone it and start it, it will give the exception

I am using Windows Platform

**Comment by HakamFostok** [comment]
Adding the following snippet `HeightRequest="100" WidthRequest="100"` solved the problem.
But still without adding this snippet it giving me an exception.
I think the readme should be updated and also how can we make the chart responsive,
I do not want to specify Height and Width, what would be the solution?

Thank you

**Comment by XceedBoucherS** [comment]
Hi,

I know why you have the exception. 
Your Chart is located inside a ScrollView. 
We will look to fix this.
In the meantime, you could remove the ScrollView, or insert you chart inside a Border with a defined height.

As for responsive chart, what are you expecting exactly ?

Thanks

**Comment by XceedBoucherS** [comment]
This should be fixed for the next release : v1.6.
Thank you

**Comment by HakamFostok** [comment]
Thank you very much, really appreciate all the efforts
